packages/plugin-essentials/sources/commands/whoami.ts

```diff
@@ -0,0 +1,61 @@
+import {Configuration, MessageName, PluginConfiguration, StreamReport} from '@berry/core';
+import {httpUtils}                                                     from '@berry/core';
+import {Clipanion}                                                     from 'clipanion';
+import {Readable, Writable}                                            from 'stream';
+
+// eslint-disable-next-line arca/no-default-export
+export default (clipanion: Clipanion, pluginConfiguration: PluginConfiguration) => clipanion
+
+  .command(`whoami`)
+  .describe(`display username`)
+
+  .detail(`
+    Print the registry username to standard output.
+  `)
+
+  .example(
+    `Print username for the default registry`,
+    `yarn whoami`,
+  )
+
+  .action(async ({cwd, stdin, stdout}: {cwd: string, stdin: Readable, stdout: Writable, registry: string}) => {
+    const configuration = await Configuration.find(cwd, pluginConfiguration);
+
+    const report = await StreamReport.start({configuration, stdout}, async report => {
+      const authorization = getAuthorizationHeader(configuration);
+
+      if (authorization === null)
+        return report.reportError(MessageName.AUTHENTICATION_NOT_FOUND, `Authentication not found`);
+
+      const headers = { authorization };
+
+      try {
+        const registryUrl = getRegistryUrl(configuration);
+        const responseBuffer = await httpUtils.get(`${registryUrl}-/whoami`, configuration, { headers });
+        const jsonResponse = JSON.parse(responseBuffer.toString());
+
+        report.reportInfo(MessageName.UNNAMED, jsonResponse.username);
+      } catch (err) {
+        if (err.statusCode === 401)
+          report.reportWarning(MessageName.UNNAMED, `Unauthorized`);
```

Comment:
That would be an error more than a warning, right? It would mean that a token is defined, but it's invalid. I guess it's such a case users would expect `yarn npm whoami` to exit with a non-zero exit code.

I think it would also need its own report code, and a slightly more verbose message. Something like this:

```suggestion
          report.reportError(MessageName.AUTHENTICATION_INVALID, `Authentication failed - your credentials may have expired`);
```